src/robot/ConcreteRobot.cpp

```diff
@@ -331,6 +331,64 @@ ConcreteRobot::getTrajectoryPostProcessor(
       feasibilityApproxTolerance);
 }
 
+//==============================================================================
+std::shared_ptr<aikido::planner::TrajectoryPostProcessor>
+ConcreteRobot::getTrajectoryPostProcessor(
+    const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+    const aikido::planner::PostProcessorParams& postProcessorParams) const
+{
+  // TODO (sniyaz):Should we  delete `smoothPath`, `retimePath`, and
+  // `retimePathWithKunz` in a subsequent PR?
+  // TODO: (sniyaz): Include custom limits in params.
+  switch (postProcessorParams.mPostProcessorType)
+  {
+    case aikido::planner::PostProcessorType::HAUSER:
+      return std::make_shared<ParabolicSmoother>(
+          getVelocityLimits(*metaSkeleton),
+          getAccelerationLimits(*metaSkeleton),
+          postProcessorParams.mHauserParams.mEnableShortcut,
+          postProcessorParams.mHauserParams.mEnableBlend,
+          postProcessorParams.mHauserParams.mShortcutTimelimit,
+          postProcessorParams.mHauserParams.mBlendRadius,
+          postProcessorParams.mHauserParams.mBlendIterations,
+          postProcessorParams.mHauserParams.mFeasibilityCheckResolution,
+          postProcessorParams.mHauserParams.mFeasibilityApproxTolerance);
+    case aikido::planner::PostProcessorType::KUNZ:
+      return std::make_shared<KunzRetimer>(
+          getVelocityLimits(*metaSkeleton),
+          getAccelerationLimits(*metaSkeleton),
+          postProcessorParams.mKunzParams.mMaxDeviation,
+          postProcessorParams.mKunzParams.mTimeStep);
+    default:
+      // TODO (sniyaz): Can this be made a compile-time check instead?
+      std::stringstream message;
+      message << "PostProcessor type not recognized in `postProcessorParams`!";
+      throw std::runtime_error(message.str());
+  }
+}
+
+//==============================================================================
+UniqueSplinePtr ConcreteRobot::postProcessPath(
+    const dart::dynamics::MetaSkeletonPtr& metaSkeleton,
+    const aikido::trajectory::Trajectory* path,
+    const constraint::TestablePtr& constraint,
+    const aikido::planner::PostProcessorParams& postProcessorParams)
+{
+  auto postProcessor
+      = getTrajectoryPostProcessor(metaSkeleton, postProcessorParams);
+
+  auto interpolated = dynamic_cast<const Interpolated*>(path);
+  if (interpolated)
+    return postProcessor->postprocess(
+        *interpolated, *(cloneRNG().get()), constraint);
+
+  auto spline = dynamic_cast<const Spline*>(path);
+  if (spline)
+    return postProcessor->postprocess(*spline, *(cloneRNG().get()), constraint);
+
+  throw std::invalid_argument("Path should be either Spline or Interpolated.");
```

Comment:
Same here. Is it possible to print to stderr and return nullptr and give the calling program a chance to recover?

Reply:
Yeah, that seems reasonable to me!